framework/src/controller/bus.js

```diff
@@ -62,12 +63,12 @@ class Bus extends EventEmitter2 {
 				.catch(error => cb(error));
 		});
 
-		return new Promise((resolve, reject) => {
-			// TODO: wait for all sockets to be created
-			this.rpcSocket.once('bind', resolve);
-			this.rpcSocket.once('error', reject);
-			this.rpcSocket.bind(socketsPath.rpc);
-		});
+		return Promise.race([
+			this._resolveWhenAllSucceed(),
+			this._rejectWhenFirstFails(),
+			// Timeout is needed here in case "bind" events never arrive
+			this._rejectWhenTimeout(2000), // TODO: Get value from config constant
+		]);
```

Comment:
Where are the event handlers from `_resolveWhenAllSucceed`, `_rejectWhenFirstFails`, and `_rejectWhenTimeout` being cleaned up?

Maybe instead of returning the `Promise.race(...)`, you can `await Promise.race(...)` then on the next line, you can unbind all events. You could add a single private method to unbind all events.

Note that this still applies even if you use `once` instead of `on` because then the handler will stay bound to the socket and could still be called unexpectedly at some point in the future.

Reply:
I think only event handlers from `_resolveWhenAllSucceed` should be removed. The other handlers from `_rejectWhenFirstFails()` will be indirectly removed as if any of those events are fired the application stops it's execution (intentionally). @lsilvs what do you think?

Reply:
After discussion with @jondubois I now remove all the listener inside a `finally` clause.